documentation/specs/Select.md

```diff
@@ -0,0 +1,216 @@
+- `isLabelVisuallyHidden` property can be used to visually hide the label.
+- Supports disabled state.
+
+### Risks and Challenges
+
+- Ensuring dropdown doesn't inadvertently get cut off.
+- Maintaining an approachable and familiar API; similar to existing Easy UI form related components.
+- Ensuring the component remains accessible when label is visually hidden.
+
+### Prior Art
+
+- [Paste `<Select />`](https://paste.twilio.design/components/select)
+- [Polaris `<Select />`](https://polaris.shopify.com/components/selection-and-input/select)
+- [Spectrum `<Picker />`](https://react-spectrum.adobe.com/react-spectrum/Picker.html)
+
+---
+
+## Design
+
+### API
+
+```ts
+import type { AriaSelectProps } from "react-aria";
+
+export type SelectSize = "sm" | "md" | "lg";
+export type ValidationState = "valid" | "invalid";
+
+export type SelectProps = AriaSelectProps & {
+  /**
+   * Visually hides the label, but keeps it accessible.
+   * @default false
+   */
+  isLabelVisuallyHidden?: boolean;
+  /**
+   * Whether the select field is disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether user input is required on the select field before form submission.
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * Whether the select field should display its "valid" or "invalid" visual styling.
+   * @default valid
+   */
+  validationState?: ValidationState;
+  /**
+   * Label text displays with emphasis.
+   * @default false
+   */
+  isLabelEmphasized?: boolean;
+  /**
+   * Whether the element should receive focus on render.
+   * @default false
+   */
+  autoFocus?: boolean;
+  /**
+   * Size affects the overall size of the select field, but it also influences
+   * the size of iconAtStart.
+   * @default md
+   */
+  size?: SelectSize;
+  /** The content to display as the label. */
+  label: ReactNode;
+  /** Error text that appears below select field. */
+  errorText?: ReactNode;
+  /** Helper text that appears below select field. */
+  helperText?: ReactNode;
+  /** Left aligned icon on select field. */
+  iconAtStart?: IconSymbol;
+  /** Method that is called when the open state of the select field changes. */
+  onOpenChange?: (isOpen: boolean) => void;
+  /** Sets the open state of the select field. */
+  isOpen?: boolean;
+  /** The currently selected key in the collection (controlled). */
+  selectedKey?: Key | null;
+  /** The initial selected key in the collection (uncontrolled). */
+  defaultSelectedKey?: Key;
+  /** Handler that is called when the selection changes. */
+  onSelectionChange?: (key: Key) => any;
+  /** The contents of the collection. */
+  children: CollectionChildren<T>;
+  /** Item objects in the collection. */
+  items?: Iterable<T>;
+  /** The item keys that are disabled. These items cannot be selected, focused, or otherwise interacted with. */
+  disabledKeys?: Iterable<Key>;
+};
+```
+
+### Anatomy
```

Comment:
my initial approach was to try to make use of the `Menu` in a way where i could get away with just defining my trigger, the select field itself. i ran into a few issues with that approach with regards to state management and aria props, and i felt the way i was needing to extend `Menu` felt unnecessarily awkward/complex. i had a look at how Spectrum approached this and it looks like they have a common `Popover` component. depending on how this implementation shakes out, it might be useful for us in the future to create some kind of generic popover. for now i think a decent path forward is to implement the `Select` by using the react-aria/react-stately select related hooks and by taking heavy inspiration from the existing `Menu` and how it handles the `OverlayContainer` and `Item`.

Reply:
that seems like a good path to me. once we have two implementations it may be easier to see how the two align

the only concern is ensuring we're not unnecessarily duplicating configuration that should change together. if the popup menu styles should be the same for the select dropdown and the menu, for instance, it would be nice to figure out how to share those configurations so they don't get out of sync

Reply:
agree